cogs/fun.py

```diff
@@ -422,45 +424,102 @@ async def bill(self, ctx):
 
 			await ctx.send(url + tag['href'])
 
-	@commands.command(hidden=True)
-	@commands.cooldown(rate=3, per=10.0, type=commands.BucketType.user)
+
+	@commands.group(name='xkcd', invoke_without_command=True)
 	@commands.bot_has_permissions(embed_links=True)
-	async def xkcd(self, ctx, *, id: int = None):
-		'''Get a random or specified xkcd comic.'''
+	async def xkcd(self, ctx, *, search : typing.Union[int, str] = None):
+		'''Gets an xkcd comic.'''
+		async with ctx.typing():
+			if search is None:
+				await self.random(ctx)
+			elif isinstance(search, int):
+				await self.num(ctx, search)
+			else:
+				await self.relevant(ctx, search=search)
+
+	@commands.cooldown(rate=3, per=10.0, type=commands.BucketType.user)
+	@xkcd.command(aliases=['id','#'])
+	async def num(self, ctx, id):
```

Comment:
Validate input for integer values only.
```suggestion
	async def num(self, ctx, id: int):
```